code/control/FreshdeskAPI.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Handler\CurlHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Request;
+
+class FreshdeskAPI extends \Object
+{
+    /**
+     * @var int
+     */
+    const maxRetries = 5;
+    /**
+     * @var array
+     */
+    private $client;
+
+    public function __construct()
+    {
+        $this->client = $this->createClient();
+    }
+
+    /**
+     * Generic method to handle requests to the Freshdesk API.
+     *
+     * @param string $method, String $url, String $action, Array $headers, Array $data
+     * @param mixed  $url
+     * @param mixed  $action
+     * @param mixed  $data
+     * @param mixed  $headers
+     *
+     * @return $response || false
+     */
+    public function call($method, $url, $action, $data = [], $headers = ['Content-type' => 'application/json'])
```

Comment:
You should always default to "private", unless there is a good reason to expose methods. `call`, `getFieldMappings`, `humanReadable` should probably be private. The reason for this is that if you mark something public, it forever becomes your public API of this class, and you will have to maintain backwards compatibility. This makes refactoring harder, and incentivises people to hack around your API.